plugins/scaffolder-backend/src/scaffolder/actions/builtin/catalog/fetchMultiple.ts

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { CatalogApi } from '@backstage/catalog-client';
+import { createTemplateAction } from '@backstage/plugin-scaffolder-node';
+import yaml from 'yaml';
+
+const id = 'catalog:fetch::multiple';
+
+const examples = [
+  {
+    description: 'Fetch entities by reference',
+    example: yaml.stringify({
+      steps: [
+        {
+          action: id,
+          id: 'fetchMultiple',
+          name: 'Fetch catalog entities',
+          input: {
+            entityRefs: ['component:default/name', 'component:default/name2'],
+            fields: ['kind', 'metadata.name'],
+          },
+        },
+      ],
+    }),
+  },
+];
+
+/**
+ * Returns entities from the catalog by entity references.
+ * @public
+ */
+export function createFetchCatalogEntitiesAction(options: {
+  catalogClient: CatalogApi;
+}) {
+  const { catalogClient } = options;
+
+  return createTemplateAction<{ entityRefs: string[]; fields?: string[] }>({
+    id,
+    description: 'Returns entities from the catalog by entity references',
+    examples,
+    schema: {
+      input: {
+        required: ['entityRefs'],
```

Comment:
Just to check this an extra round, did you consider adding this option to the existing `catalog:fetch` action? I'm thinking something like an additional `entityRefs` option that can be set instead of `entityRef`, paired with an `entities` output that will either be just the single entity that was fetch, or the full list of entities when using `entityRefs`.

Reply:
That might be a good option as well, I can refactor 👍